python/pmtiles/convert.py

```diff
@@ -178,3 +178,167 @@ def pmtiles_to_dir(input, output):
             os.makedirs(directory, exist_ok=True)
             with open(path, "wb") as f:
                 f.write(tile_data)
+
+
+def disk_to_pmtiles(directory_path, output, maxzoom, **kwargs):
+    """Convert a directory of tiles on disk to PMTiles.
+
+    Requires metadata.json in the root of the directory.
+
+    Tiling scheme of the directory is assumed to be zxy unless specified.
+
+    Args:
+        directory_path (str): Root directory of tiles.
+        output (str): Path of PMTiles to be written.
+        maxzoom (int, "auto"): Max zoom level to use. If "auto", uses highest zoom in directory.
+
+    Keyword args:
+        scheme (str): Tiling scheme of the directory ('ags', 'gwc', 'zyx', 'zxy' (default)).
+        tile_format (str): Image format of the tiles ('png', 'jpeg', 'pbf', 'webp', 'avif') if not given in the metadata.
+        verbose (bool): Set True to print progress.
+
+    Uses modified elements of 'disk_to_mbtiles' from mbutil
+
+    Copyright (c), Development Seed
+    All rights reserved.
+
+    Licensed under BSD 3-Clause
+    """
+    verbose = kwargs.get("verbose")
+    try:
+        metadata = json.load(open(os.path.join(directory_path, 'metadata.json'), 'r'))
+    except IOError:
+        raise Exception("metadata.json not found in directory")
+
+    tile_format = kwargs.get('tile_format', metadata.get("format"))
+    if not tile_format:
+        raise Exception("tile format not found in metadata.json nor specified as keyword argument")
+    metadata["format"] = tile_format  # Add 'format' to metadata
+
+    scheme = kwargs.get('scheme')
+
+    # Collect a set of all tile IDs
+    z_set = []  # List of all zoom levels for auto-detecting maxzoom.
+    tileid_path_set = []  # List of tile (id, filepath) pairs
+    zoom_dirs = get_dirs(directory_path)
+    zoom_dirs.sort(key=len)
+    try:
+        collect_max = int(maxzoom)
+    except ValueError:
+        collect_max = 99
+    collect_min = metadata.get("minzoom", 0)
```

Comment:
What tool is creating metadata.json? https://github.com/felt/tippecanoe writes the minzoom and maxzoom as strings, so these may need to be `int(metadata.get...))`

example tippecanoe command: `./tippecanoe --output-to-directory foo ne_10m_admin_0_label_points.fgb --maximum-zoom 5 --force`

Reply:
tippecanoe also writes a key called `json` that has `vector_layers`. I think this code will need to special-case `format=pbf` in order to translate `vector_layers` correctly as they are required in the spec:

https://github.com/protomaps/PMTiles/blob/main/spec/v3/spec.md#5-json-metadata

The other key it writes is `tilestats` - so all top-level keys of the `json` value could lifted up one level in the pmtiles metadata.

Tools other than tippecanoe might write different `metadata.json`?

Reply:
If the above sounds like overkill then maybe we limit the disk_to_pmtiles to raster formats only (if that's the use case you need)

Reply:
Ah I wasn't aware of that. Thanks for the explanation. 

For now let's limit it to raster formats. That's indeed the only use case I need. If someone with a need (and knowledge) for other formats wishes to add support, there's at least this function for them to add to.

Reply:
Are you OK if I simply remove 'pbf` from the help and docstrings of the functions?

Or would you like an explicit rejection for formats not 'png', 'jpeg', 'webp', or 'avif'?

Reply:
removing `pbf` is fine.

Reply:
References to 'pbf' are now gone, and I added 'raster format' to be more specific.